mdagent/tools/base_tools/analysis_tools/salt_bridge_tool.py

```diff
@@ -0,0 +1,106 @@
+from typing import Optional
+
+import mdtraj as md
+from langchain.tools import BaseTool
+from pydantic import BaseModel, Field
+
+from mdagent.utils import PathRegistry
+
+
+class SaltBridgeFunction:  # this class defines a method called find_salt_bridge
+    # using MD traj and top files and threshold distance default, residue pair list
+    # used to account for salt bridge analysis
+    def __init__(self, path_registry):
+        self.path_registry = path_registry
+        self.salt_bridges = []  # stores paired salt bridges
+        self.traj = None
+
+    def find_salt_bridges(
+        self, traj_file, top_file=None, threshold_distance=0.4, residue_pairs=None
+    ):
+        # add two files here in similar format as line 14 above
+        traj_file_path = self.path_registry.get_mapped_path(traj_file)
+        ending = traj_file_path.split(".")[-1]
+        if ending in ["dcd", "xtc", "xyz"] and top_file is not None:
+            top_file_path = self.path_registry.get_mapped_path(top_file)
+            self.traj = md.load(traj_file_path, top=top_file_path)
+        else:
+            self.traj = md.load(traj_file_path)
```

Comment:
can you use the helper function in mdagent/utils/data_handling.py `load_single_traj` instead? keep it in a try except block though